misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/FunctionMock.java

```diff
@@ -186,38 +196,48 @@ private static void validateFunctionSignature(Method mockMethod, Method original
+                        null,
+                        new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
             }
 
             // Validate if param number is the same
             if (mockMethodParameters.length != originalMethodParameters.length) {
-                throw ErrorCreator.createDistinctError(MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
-                                                       MockConstants.TEST_PACKAGE_ID, StringUtils.fromString(
+                throw ErrorCreator.createError(
+                        MockConstants.TEST_PACKAGE_ID,
+                        MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
+                        StringUtils.fromString(
                                 "Parameter types of function " + mockMethod.getName() +
-                                        " does not match function" + originalMethod.getName()));
+                                        " does not match function" + originalMethod.getName()),
+                        null,
+                        new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
             }
 
             // Validate each param
             for (int i = 0; i < mockMethodParameters.length; i++) {
                 if (mockMethodParameters [i] != originalMethodParameters[i]) {
-                    throw ErrorCreator.createDistinctError(MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
-                                                           MockConstants.TEST_PACKAGE_ID, StringUtils.fromString(
+                    throw ErrorCreator.createError(
+                            MockConstants.TEST_PACKAGE_ID,
+                            MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
+                            StringUtils.fromString(
                                     "Parameter types of function " + mockMethod.getName() +
-                                            "does not match function" + originalMethod.getName()));
+                                            "does not match function" + originalMethod.getName()), null,
+                            new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
                 }
             }
 
         } else {
-            throw ErrorCreator.createDistinctError(MockConstants.FUNCTION_NOT_FOUND_ERROR,
-                                                   MockConstants.TEST_PACKAGE_ID,
-                                                   StringUtils.fromString(
-                                                            "Mock function \'" + mockMethodName + "\' " +
-                                                                    "cannot be found"));
+            throw ErrorCreator.createError(
+                    MockConstants.TEST_PACKAGE_ID,
+                    MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
+                    StringUtils.fromString("Mock function \'" + mockMethodName + "\' " +
```

Comment:
```suggestion
                    StringUtils.fromString("Mock function '" + mockMethodName + "' " +
```
Why do we need to escape single quotes?

Reply:
Removed escaped single quotes